### ISSUE TITLE
testmon sometimes runs too many tests if the test filename ends in "_test"

### ISSUE THREAD

**Issue by tntmarket** [tarpas/pytest-testmon#91]
In my project if I:

1. run testmon for the first time => it runs all the tests as expected
2. change a test case and run testmon again => it only runs that test case, as expected
3. change the same test case and run testmon again => it runs too many tests

![testmon sometimes runs all tests](https://media.giphy.com/media/2bUoNab9OcCdoAVKmp/giphy.gif)

This example runs all tests on the 3rd time. My project has ~300 tests, and it seems to run an arbitrary number of tests each time if I change/revert the same file.

I noticed that if I rename `letter_test.py` => `test_letter.py`, the problem seems to go away.

Furthermore, [this test case failed](https://github.com/tarpas/pytest-testmon/compare/master...tntmarket:patch-1) after I flipped the filenames:

```

test/test_testmon.py:275 (TestmonDeselect.test_new)
self = <test.test_testmon.TestmonDeselect object at 0x102018828>
testdir = <Testdir local('/private/var/folders/nh/8gy_s8g54m98qnl2kvmzws2rp11zsb/T/pytest-of-davelu/pytest-34/test_new0')>

    def test_new(self, testdir):
        a = testdir.makepyfile(a="""
                def add(a, b):
                    a = a
                    return a + b
    
                def subtract(a, b):
                    return a - b
            """)
    
        testdir.makepyfile(b="""
                def divide(a, b):
                    return a // b
    
                def multiply(a, b):
                    return a * b
            """)
    
        testdir.makepyfile(a_test="""
                from a import add, subtract
                import time
    
                def test_add():
                    assert add(1, 2) == 3
    
                def test_subtract():
                    assert subtract(1, 2) == -1
                        """)
    
        testdir.makepyfile(b_test="""
                import unittest
    
                from b import multiply, divide
    
                class TestB(unittest.TestCase):
                    def test_multiply(self):
                        self.assertEqual(multiply(1, 2), 2)
    
                    def test_divide(self):
                        self.assertEqual(divide(1, 2), 0)
            """)
    
    
        testdir.makepyfile(ab_test="""
                from a import add
                from b import multiply
                def test_add_and_multiply():
                    assert add(2, 3) == 5
                    assert multiply(2, 3) == 6
            """)
        result = testdir.runpytest("--testmon", )
        result.stdout.fnmatch_lines([
            "*5 passed*",
        ])
        result = testdir.runpytest("--testmon")
        result.stdout.fnmatch_lines([
            "*collected 0 items*",
        ])
        a.setmtime(1424880935)
        result = testdir.runpytest("--testmon")
        result.stdout.fnmatch_lines([
>           "*5 deselected*",
        ])

/Users/davelu/workspace/pytest-testmon/test/test_testmon.py:337: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <_pytest.pytester.LineMatcher object at 0x101f62f98>
lines2 = ['*5 deselected*']

    def fnmatch_lines(self, lines2):
        """Search captured text for matching lines using ``fnmatch.fnmatch``.
    
            The argument is a list of lines which have to match and can use glob
            wildcards.  If they do not match a pytest.fail() is called.  The
            matches and non-matches are also printed on stdout.
    
            """
>       self._match_lines(lines2, fnmatch, 'fnmatch')
E       Failed: nomatch: '*5 deselected*'
E           and: '============================= test session starts =============================='
E           and: 'platform darwin -- Python 3.6.4, pytest-3.4.1, py-1.5.2, pluggy-0.6.0'
E           and: 'testmon=True, changed files: 0, skipping collection of 0 items, run variant: default'
E           and: 'rootdir: /private/var/folders/nh/8gy_s8g54m98qnl2kvmzws2rp11zsb/T/pytest-of-davelu/pytest-34/test_new0, inifile:'
E           and: 'plugins: testmon-0.9.10'
E           and: 'collected 5 items'
E           and: ''
E           and: 'a_test.py ..                                                             [ 40%]'
E           and: 'ab_test.py .                                                             [ 60%]'
E           and: 'b_test.py ..                                                             [100%]'
E           and: ''
E           and: '=========================== 5 passed in 0.06 seconds ==========================='
E           and: ''
E       remains unmatched: '*5 deselected*'
```

**Comment by tarpas** [comment]
Thanks for report. The feature of taking into account 'python_files' ini option is missing since the beginning. 
https://docs.pytest.org/en/latest/customize.html#confval-python_files

'test_*' is hardcoded here:
https://github.com/tarpas/pytest-testmon/blob/master/testmon/pytest_testmon.py#L213

Also the code moved in my recent effort for code improvements.

I hope to have time to fix it by the end of next week.

**Comment by tarpas** [comment]
@tntmarket Could you try unreleased version? d9083c43b4e266771dbb59c1a5cfb8e4f96f2a2b
It passes your test but has a lot of changes and I didn't test it in real usage :/

**Comment by tarpas** [comment]
Should be fixed with v0.9.10
